st2tests/st2tests/fixtures/aliases/BUILD

```diff
@@ -1 +1,7 @@
-python_sources()
+pack_metadata(
+    name="metadata",
```

Comment:
Slightly confused. In the dummy_pack_23 we don't include the pack_metadata as there is no pack.yaml in that pack.

For things like this BUILD file and others at this level in the hierarchy, why do they have the pack_metatdata?

Reply:
`pack_metadata` includes more than just `pack.yaml`.
https://github.com/StackStorm/st2/blob/a73225c3a416c19a3dc1a0edf51ff48e2acf5202/pants-plugins/pack_metadata/target_types.py#L27-L40

It also includes metadata yaml files for resources that go in packs: actions, aliases, sensors, etc.

Several of the directories at this level do not have `pack.yaml`, but they do have the other metadata yaml files. Examples:

- [st2tests/st2tests/fixtures/aliases/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/aliases)
    - [actions/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/aliases/actions/)
        - [action1.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/actions/action1.yaml)
        - [action2.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/actions/action2.yaml)
        - [action3.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/actions/action3.yaml)
        - [action4.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/actions/action4.yaml)
    - [aliases/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/aliases/aliases/)
        - [alias1.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/aliases/alias1.yaml)
        - [alias2.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/aliases/alias2.yaml)
        - [alias3.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/aliases/alias3.yaml)
        - [alias4.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/aliases/alias4.yaml)
        - ...
    - [runners/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/aliases/runners/) _runners are not typically defined in packs, but close enough._
        - [runner1.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/aliases/runners/runner1.yaml)
- [st2tests/st2tests/fixtures/backstop/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/backstop)
    - [rules/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/backstop/rules/)
        - [backstop.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/backstop/rules/backstop.yaml)
        - [fail.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/backstop/rules/fail.yaml)
        - [success.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/backstop/rules/success.yaml)
    - [triggers/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/backstop/triggers/) _this is not common in packs, but *is* allowed._
        - [trigger1.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/backstop/triggers/trigger1.yaml)
    - [triggertypes/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/backstop/triggertypes/) _this is even less common, and I'm not sure if it can be defined in packs, but close enough._
        - [triggertype1.yaml](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/backstop/triggertypes/triggertype1.yaml)
- [st2tests/st2tests/fixtures/descendants/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/descendants) _this is not like a pack, but has db entries in the form of yaml files which is sort of pack-ish_
    - [executions/](https://github.com/StackStorm/st2/tree/master/st2tests/st2tests/fixtures/descendants/executions/) _this would never be defined in a pack_
        - [child1_level1](https://github.com/StackStorm/st2/blob/master/st2tests/st2tests/fixtures/descendants/executions/child1_level1.yaml)
        - ...

Then the other fixtures do similar things. They all seem to have mongo docs in the form of yaml files (after-all the db effectively just holds all of the metadata files in json format, so going in the reverse makes sense).

We need `resources` for all of these yaml files. `PackMetadata` subclasses `Resources`, so everything in the `pack_metadata` target also counts as a resource. So, we either need to:
- add `resources(sources=...)` targets that encompass all of the yaml files, and make the python fixture.py depend on them
- add `pack_metadata` and make the python fixture.py depend on it

It was quick and easy to just treat these as packs using `pack_metadata` so that's what I did. Would it make more sense to you if we used `resources()` instead for these?

Reply:
It's fine - thanks for clarification. 
My point for not reading the resource correctly / I did go and look and must have not scrolled past the pack example yaml line.

All good!